docs: refactor README, CONTRIBUTING (#179)
openebs/community: contribute/labels-of-issues.md

| Label | Description |
|---|---|
| kind/api  | Issue/PR related to API  |
| kind/bug |  If there’s an error, failure in workflow, fault etc.|
| kind/feature  | New / existing / required feature or enhancement |
| kind/improve | Needs improvement |
| kind/e2e-test | Need/Add end to end test |
| kind/unit-test | Need/Add Unit test|
| kind/security  | Security-related concerns  |
| kind/testing  |  Needs/Adds tests |
| kind/good-first  |  beginner friendly |

---

## Help

Labels prefixed with *help* implies that issue needs some help from community/users. Further, you can adjust the level of help required, by specifying small, medium, large etc.

| Label | Description |
|---|---|
| help/small  | Need small help on docs (beginner level)|
| help/medium | Need help to fix the code ( Intermediate level) |
| help/large | Need help on design changes and code review (Expert level) |

---

## Area

Labels prefixed with *area* indicates that which area issue or PR belong. Here, area implies component of the OpenEBS project. It will be easier to sort an issue based on the area of user's interest.

| Label | Description |
|---|---|

> | area/maya  | Maya Specific  |

Reviewer: ```suggestion
| area/networking  | Networking Specific  |
```

| area/k8s | Kubernetes specific   |
| area/ci  | Continuous integration specific  |
| area/high-availability  | High availability  |
| area/storage | Storage specific  |
| area/machine-learning  |  Machine learning specific |

---

## Size

> Labels prefixed with *size* defines the size of change done in particular PR.

Reviewer: sounds like this is saying it's the size on a PR
but the label should be added on a issue, which estimates what size a future PR would eventually be?

Author: I think it's that, once decided for pickup, someone can add the size on the issue?

Reviewer: Then slightly reworking as to me it's reading the size already done rather than an estimate

```suggestion
Labels prefixed with *size* defines the expected size of the changes required to fix the issue.
```

| Label | Description |
|---|---|
| size/L | Size of change is large/7 commits probably |
| size/M | Size of change is medium/5 commits probably |
| size/S | Size of change is small/3 commits probably |
| size/XL | Size of change is extra large/10 commits probably |
| size/XS | Size of change is extra small/1 commit probably |
| size/XXL | Size of change is very large/15 commits probably |

### NOTE

> Please do not club major changes into single PR. It'll be difficult for a reviewer to review a big PR. In that case, reviewer may discard your PR and will ask you create multiple PRs.

Reviewer 2: a single PR

Reviewer 2: to create multiple PRs.

---

### Priority

Labels with prefix *priority* defines the severity of the issue. Severity level of an issue can be given as a suffix.

| Label | Description |
|---|---|

> | priority/0 | Urgent: Must be fixed in the next build. |

Reviewer: kubernetes has these in words: https://github.com/kubernetes/kubernetes/labels?q=priority+

Author: This looks better to me. Replaced.

| priority/1 | High: Must be fixed in any of the upcoming builds but should be included in the release. |
| priority/2 | Medium: Maybe fixed after the release / in the next release. |
| priority/3 | Low: May or may not be fixed at all. |

---

### Sub-Project

Labels with prefix *repo* points the subproject where given issue needs to be fixed. If users have no idea where to file the issue then they can file a issue in `openebs/openebs` later OpenEBS contributors/authors/owners will tag issue with the appropriate repository using *repo* label.

| Label | Description |
|---|---|
| subproject/mayastor | Related to Replicated PV Mayastor and its components |
| subproject/zfs-localpv | Related to ZFS Local PV and its components |
| subproject/lvm-localpv | Related to LVM Local PV and its components |

> | subproject/hostpath-localpv |  Related to Hostpath Local PV and its components |

Reviewer: dynamic-localpv or dynamic-provisioner-localpv?
In case we actually add a driver called hostpath in the future (eg: unified rawfile+dynamic)
```suggestion
| subproject/dynamic-localpv |  Related to Hostpath Local PV and its components |
```

Author: dynamic-localpv in that case sounds more appropriate. Done!

This document provides a list of labels and their purpose. These labels can be attached to issues and PRs. The aim behind having such list of labels is, we can sort and distinguish issues. This list will help new community members to dive-in OpenEBS if they wish to contribute to project or investigate the problem.

Use case:
> I am a beginner, I want to contribute to OpenEBS project and I am not sure from where to begin?

Answer:
> Start looking into the issues tagged with labels such as `help/small` , `size/XS`, `kind/unit-test` etc. Solve such issues and raise PR.

---

## Kind

Labels prefixed with *kind* indicate the type of issue and PR. Suppose, you want to report a bug then you can tag your issue with `kind/bug` label.

| Label | Description |
|---|---|
| kind/api  | Issue/PR related to API  |
| kind/bug |  If there’s an error, failure in workflow, fault etc.|
| kind/feature  | New / existing / required feature or enhancement |
| kind/improve | Needs improvement |
| kind/e2e-test | Need/Add end to end test |
| kind/unit-test | Need/Add Unit test|
| kind/security  | Security-related concerns  |
| kind/testing  |  Needs/Adds tests |

> | kind/good-first  |  beginner friendly |

Reviewer 3: Are we not using labels similar to https://github.com/openebs/mayastor/labels?

Author: Let's leave out the label discussion out of this PR. I will raise a different one once we all decide.

| Label | Description |
|---|---|
| help/small  | Need small help on docs (beginner level)|
| help/medium | Need help to fix the code ( Intermediate level) |
| help/large | Need help on design changes and code review (Expert level) |

---

## Area

Labels prefixed with *area* indicates that which area issue or PR belong. Here, area implies component of the OpenEBS project. It will be easier to sort an issue based on the area of user's interest.

| Label | Description |
|---|---|
| area/networking  | Networking Specific  |
| area/k8s | Kubernetes specific   |
| area/ci  | Continuous integration specific  |
| area/high-availability  | High availability  |
| area/storage | Storage specific  |
| area/machine-learning  |  Machine learning specific |

---

## Size

Labels prefixed with *size* defines the size of change done in particular PR.

| Label | Description |
|---|---|
| size/L | Size of change is large/7 commits probably |
| size/M | Size of change is medium/5 commits probably |
| size/S | Size of change is small/3 commits probably |
| size/XL | Size of change is extra large/10 commits probably |
| size/XS | Size of change is extra small/1 commit probably |

> | size/XXL | Size of change is very large/15 commits probably |

Reviewer: update with the org size definitions please

Author: Done.

| Label | Description |
|---|---|
| priority/backlog | Higher priority than priority/awaiting-more-evidence. |
| priority/critical-urgent | Highest priority. Must be actively worked on as someone's top priority right now. |
| priority/important-longterm | Important over the long term, but may not be staffed and/or may need multiple releases to complete. |
| priority/important-soon | Must be staffed and worked on either currently, or very soon, ideally in time for the next release. |
| priority/awaiting-more-evidence | Lowest priority. Possibly useful, but not yet enough support to actually get it done. |

---

### Sub-Project

Labels with prefix *repo* points the subproject where given issue needs to be fixed. If users have no idea where to file the issue then they can file a issue in `openebs/openebs` later OpenEBS contributors/authors/owners will tag issue with the appropriate repository using *repo* label.

| Label | Description |
|---|---|
| subproject/mayastor | Related to Replicated PV Mayastor and its components |
| subproject/zfs-localpv | Related to ZFS Local PV and its components |
| subproject/lvm-localpv | Related to LVM Local PV and its components |

> | subproject/dyamic-localpv |  Related to Hostpath Local PV and its components |

Reviewer: ```suggestion
| subproject/dynamic-localpv |  Related to Hostpath Local PV and its components |
```

Although, after last discussion should we rename these now to "localpv-{zfs,lvm,rawfile,hostpath}" ?

Author: Okay

| subproject/rawfile-localpv|  Related to Rawfile Local PV and its components |

---

### General

As the name suggests, this contains general category labels and they are self-explanatory.

| Label | Description |
|---|---|

> | architecting | If something needs design changes/brainstorming |

Reviewer: Update as per the org labels now please

Author: There are a lot of them, I think we should first trim down a bit? Not sure which ones to put here. Also I see a lot of labels in org which are not listed.

Author: Let's leave out the label discussion out of this PR. I will raise a different one once we all decide.
